Fix two bugs in removeVertex's private methods

shakareem opened the pull request:
When iterating over a collection concurrent modification exception is thrown if someone tried to remove an element from the collection. the solution is creating a [mutable iterator](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/-mutable-iterator/) manually. Also there was an option to use removeif which does the same, but in this case we would iterate over edges twice.

shakareem committed:
fix: removal of an element in a for loop

qrutyy reviewed (approved):
I like it!!! Good job)

shakareem commented:
Will add fix of another bug mentioned in #21 PR to this PR.

spisladqo reviewed (approved):
![image](https://github.com/spbu-coding-2023/graphs-graph-2/assets/136486754/4754d68a-da85-4dad-98a4-f51ad52b9440)

shakareem committed:
fix: make vertex's id var and simplify removeVertex method

The id was val, which made copying last added vertex to the place of removed one very complex

qrutyy reviewed (approved):
SLONYARA

spisladqo reviewed (approved):
Very very good

shakareem commented:
> SLONYARA

![IMG_20240516_230536_248~2.jpg](https://github.com/spbu-coding-2023/graphs-graph-2/assets/144351053/e118ddad-5f18-4329-b749-97f8dc181cae)

shakareem commented:
> Very very good

![800px-Duke_(Java_mascot)_waving.svg.png](https://github.com/spbu-coding-2023/graphs-graph-2/assets/144351053/e277eeda-f9cf-45c8-8799-29d4857deeb3)
